src/framework/input/element-input.js

```diff
@@ -736,15 +736,23 @@ class ElementInput {
         if (eventType === 'mouseup' && this._pressedElement) {
             // click event
             if (this._pressedElement === this._hoveredElement) {
-                this._pressedElement = null;
-
-                // fire click event if it hasn't been fired already by the touchup handler
-                if (!this._clickedEntities || !this._clickedEntities[this._hoveredElement.entity.getGuid()]) {
+                // fire click event if it hasn't been fired already by the touchend handler
+                const guid = this._hoveredElement.entity.getGuid();
+                // Always fire, if there are no clicked entities
+                let fireClick = !this._clickedEntities;
```

Comment:
```suggestion
                let fireClick = this._clickedEntities.length > 0;
```
May be a personal style but if it's an array, I prefer to check it like an array rather than `!` as I see that as checking against null/undefined

Reply:
It is checking if `this._clickedEntities` exists first, because it's not definitely assigned in the constructor:
```js
if (platform.touch)
            this._clickedEntities = {};
```

Reply:
Oops, thats' true! My bad